src/modules/navigator/mission_base.cpp

```diff
@@ -1377,3 +1378,16 @@ bool MissionBase::canRunMissionFeasibility()
 	       (_geofence_status_sub.get().geofence_id == _mission.geofence_id) &&
 	       (_geofence_status_sub.get().status == geofence_status_s::GF_STATUS_READY);
 }
+
+void MissionBase::updateMissionAltAfterHomeChanged()
+{
+	if (_navigator->get_home_position()->update_count > _home_update_counter) {
+		float altitude_diff = _navigator->get_home_position()->alt - _last_home_alt;
```

Comment:
```suggestion
		float new_alt = get_absolute_altitude_for_item(_mission_item);
		float altitude_diff = new_alt - _navigator->get_position_setpoint_triplet()->current.alt;
```

If you do it like this, you shouldn't need the _last_home_alt variable. Also it should be fine as well when the mission item is in global coordinates.

Reply:
Thank you, I have made the modifications according to your suggestion.